src/Subscriptions/SubscriptionsDataStoreCPT.php

```diff
@@ -507,19 +507,20 @@ private function update_post_meta( $subscription ) {
 		$this->update_meta( $id, 'email', ( null === $customer ? null : $customer->get_email() ) );
 		$this->update_meta( $id, 'end_date', $subscription->get_end_date() );
 		$this->update_meta( $id, 'next_payment', $subscription->get_next_payment_date() );
-		$this->update_meta( $id, 'next_payment_delivery_date', $subscription->get_next_payment_delivery_date() );
 		$this->update_meta( $id, 'version', $subscription->get_version() );
 
 		// Maybe delete next payment date post meta.
 		if ( null === $subscription->get_next_payment_date() ) {
 			\delete_post_meta( $id, $this->meta_key_prefix . 'next_payment' );
-			\delete_post_meta( $id, $this->meta_key_prefix . 'next_payment_delivery_date' );
 		}
-
+		
 		if ( null === $subscription->get_end_date() ) {
 			\delete_post_meta( $id, $this->meta_key_prefix . 'end_date' );
 		}
 
+		// Delete legacy next payment delivery date post meta.
+		\delete_post_meta( $id, $this->meta_key_prefix . 'next_payment_delivery_date' );
+
```

Comment:
This might be something for an update script, but I would skip it, some old unused post meta isn't a disaster.